main.tf

```diff
@@ -368,7 +372,7 @@ resource "aws_ecs_service" "service" {
   cluster                            = var.cluster_id
   task_definition                    = aws_ecs_task_definition.task.arn
   desired_count                      = var.desired_count
-  launch_type                        = var.launch_type
+  launch_type                        = var.use_spot ? null : var.launch_type
```

Comment:
The launch_type has a default-value of "FARGATE", but I could set up a precondition step to check if it's something else than FARGATE and fail if `use_spot` is set in the same config. 

Do you want extra checks like this in yhour modules or do you prefer to keep them cleaner without it?

Reply:
something like this with a depends_on in the service. It's not perfect though, since the user could specify "FARGATE" as the launch_type and it wouldn't fail. So the question is if you want it or not

```terraform
resource "terraform_data" "no_launch_type_and_spot" {
  lifecycle {
    precondition {
      condition     = !var.use_spot || var.launch_type == "FARGATE"
      error_message = "use_spot and launch_type are mutually exclusive"
    }
  }
}
```

Reply:
Yes, a precondtion would be great 🙏 Remember to clamp the versin to greater than 1.5 in that case

Reply:
Added the precondition!

I think they came in 1.2 though, so it should be fine with 1.3 as it is at the moment?

Reply:
1.4 it is! to support terraform_data